crates/weaver_checker/src/lib.rs

```diff
@@ -76,6 +83,16 @@ impl Error {
     }
 }
 
+/// Handles a list of errors and returns a compound error if the list is not
+/// empty or () if the list is empty.
+pub fn handle_errors(errors: Vec<Error>) -> Result<(), Error> {
```

Comment:
I have an idea for  how to make this method a bit more generic. 

TL;DR; I think we could use a `trait`  on specific `Error` types, with a macro.

So you could do something like:

enum Error {
 ...,
 
 #[compound_error]
 CompoundError(errors: Vec<Error>)
}
```

And then we'd have a method like:

```
fn handle_errors<E: WeaverError>(errors:  Vec<E>) -> Result<(), E>
```

Reply:
Yes interesting.